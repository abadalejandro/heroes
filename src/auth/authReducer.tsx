export interface IAuthState {
    name: string | null,
    logged: boolean,
}

export const initialState: IAuthState = {
    name: 'Alex',
    logged: false,
}

type LoginPayload = {
    name: string | null;
}

export type AuthAction =
    | { type: 'login', payload: LoginPayload }
    | { type: 'logout' }



export const authReducer = (state: IAuthState = initialState, action: AuthAction): IAuthState => {
    switch (action.type) {
        case 'login':
            return {
                ...action.payload,
                logged: true
            };

        case 'logout':
            return {
                ...state,
                name: null,
                logged: false
            }

        default:
            return state;
    }
}