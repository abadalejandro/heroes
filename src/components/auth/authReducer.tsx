import { types } from "../types/types";

// interface state {
//     name:string,
//     logged:boolean,
// }

// const initialState:state = {
//     name: 'Alejandro',
//     logged: true,
// }

export const authReducer = (state = {}, action: any) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                ...state,
                logged: false
            }



        default:
            return state;
    }
}