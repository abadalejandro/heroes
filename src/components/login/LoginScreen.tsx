
export const LoginScreen = () => {
    return (
        <div>
            <h1>Loginscreen</h1>
        </div>
    )
}
