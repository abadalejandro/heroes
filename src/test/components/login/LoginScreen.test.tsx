import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';

const history = {
    replace: jest.fn()
}

const contextValue = {
    dispatch: jest.fn(),
    user: { logged: false }
}

const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
        <LoginScreen history={history} />
    </AuthContext.Provider>);

describe('Test in <LoginScreen />', () => {

    test('Should show it correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should do the dispatch and navigation', () => {
        const handleClick:any = wrapper.find('button').prop('onClick');
        handleClick();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: 'login', payload: {name: 'ale'}
        });
        expect(history.replace).toHaveBeenCalledWith('/');
        localStorage.setItem('lastPath', '/dc');
        handleClick();
        expect(history.replace).toHaveBeenCalledWith('/dc');

    });

})
