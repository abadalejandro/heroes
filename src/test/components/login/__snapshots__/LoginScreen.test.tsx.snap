// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen /> Should show it correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Loginscreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
